RxSwift/Traits/Single.swift

```diff
@@ -213,6 +213,18 @@ extension PrimitiveSequenceType where Trait == SingleTrait {
             return Single(raw: self.primitiveSequence.source.map(transform))
     }
     
+    /**
```

Comment:
We can not add `compactMap` to `Single` because it breaks the contact.

Reply:
That's why it returns a `Maybe`, just like `filter` does.
Don't you think it's right?

Reply:
Oh, that's fine. I've misread that you are returning `Single`. My bad.